ci/build_container/build_recipes/gperftools.sh

```diff
@@ -7,5 +7,12 @@ VERSION=2.6.1
 wget -O gperftools-$VERSION.tar.gz https://github.com/gperftools/gperftools/releases/download/gperftools-$VERSION/gperftools-$VERSION.tar.gz
 tar xf gperftools-$VERSION.tar.gz
 cd gperftools-$VERSION
+
+if [[ `uname` == "Darwin" ]];
+then
+  # enable ucontext(3)
+  export CPPFLAGS="-D_XOPEN_SOURCE=1"
```

Comment:
For `_XOPEN_SOURCE=700` or `_XOPEN_SOURCE=600`, gperftools fails to compile because `getpagesize` is not defined in the version of `_POSIX_C_SOURCE` that those definitions imply.

`getpagesize` is defined when `_XOPEN_SOURCE=500`, but `syscall` isn't. However, defining `_DARWIN_C_SOURCE` brings `syscall` back.